ui-participant/src/studies/enroll/PreEnroll.tsx

```diff
@@ -33,9 +34,13 @@ export default function PreEnrollView({ enrollContext, survey }:
   const navigate = useNavigate()
   // for now, we assume all pre-screeners are a single page
   const pager = { pageNumber: 0, updatePageNumber: () => 0 }
+
+  const { preFilledAnswers, referralSource, clearStoredEnrollmentParams } = useEnrollmentParams()
```

Comment:
I'm a bit wary of this not being persisted in the URL. I think most of our redirects right now don't preserve query params, so it makes sense, but I do feel like it could make some situations harder to debug (+ harder to track with mixpanel). It works well as-is, though, so I'm happy to 👍 !